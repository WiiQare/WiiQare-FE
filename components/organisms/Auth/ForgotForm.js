import React, { useState } from "react";
import TextField from "@mui/material/TextField";
import Box from "@mui/material/Box";
import Button from "@mui/material/Button";
import Stack from "@mui/material/Stack";
import useMediaQuery from "@mui/material/useMediaQuery";
import { useMutation } from 'react-query';

import OnboardingScreen from "../../molecules/OnboardingScreen";
import MenuHolder from "../../atoms/MenuHolder";
import BlinkSnackbar from "../../atoms/BlinkSnackbar";
import { emailValidate } from "../../../lib/validate";
import { sendEmailReset } from "../../../lib/helper";
import Toast from "../../atoms/Toast";
import { useFormik } from "formik";
import LoadingButton from "../../atoms/Loader/LoadingButton";
import Image from "next/image";

function ForgotForm() {
	const matches = useMediaQuery("(max-width: 992px)");
	const [onboardingIsEnd, setOnboardingIsEnd] = React.useState("");
	const [state, setState] = useState({ type: 0, message: '' });
	const [complete, setComplete] = useState(false);


	React.useEffect(() => {
		setOnboardingIsEnd(localStorage.getItem("onboardingIsEnd") || "");
	}, []);

	const sendEmailMutation = useMutation(sendEmailReset, {
		onSuccess: (res) => {

			if (!res.code) {
				setState({ type: 1, message: "Mail envoyé avec succès" })
				setComplete(true)
			} else {
				setState({ type: 2, message: res.message ?? res.description })
				setTimeout(() => {
					setState({ type: 0, message: "" })
				}, 3000);
			};
		}
	});

	const onSubmit = async (values) => {
		if (Object.keys(values).length == 0) return console.log("Pas de données");
		sendEmailMutation.mutate(values)
	};

	const closeToast = () => {
		setState({ type: 0, message: "" })
	}

	// Formik hook
	const formik = useFormik({
		initialValues: {
			email: ''
		},
		validate: emailValidate,
		onSubmit
	})

	const formHolder = (
		<>
			<div className="form-holder">
				<MenuHolder href="/register" label="S'INSCRIRE" />
				<div className="signin-signup-form">
					<div className="flex flex-col md:w-3/4 w-full border-opacity-50 gap-4">

						<div className="form-items !w-full !max-w-full flex flex-col items-center mx-auto">

							{state.type > 0 ? state.type == 2 ? <Toast type={"danger"} message={state.message} close={closeToast} /> : (state.type == 1 ? <Toast type={"success"} message={state.message} close={closeToast} /> : <></>) : <></>}

							<div className="md:w-3/4 mx-auto">
								{
									!complete ? (
										<>
											<div className="form-title">Mot de passe oublié</div>
											<Box sx={{ mb: 3, mt: 2 }}></Box>
											<form id="signinform" className="" onSubmit={formik.handleSubmit}>
												<Stack spacing={2}>
													<div className="space-y-1">

														<TextField
															id="outlined-basic"
															fullWidth
															label="Adresse email"
															variant="outlined"
															name="email"
															{...formik.getFieldProps('email')}
														/>
														{formik.errors.email && formik.touched.email ? <span className="flex items-center gap-1 text-rose-500 text-left text-xs px-1"><HiOutlineInformationCircle /><span>{formik.errors.email}</span></span> : <></>}
													</div>

													<Box>
														<Button
															size="large"
															variant="contained"
															type="submit"
															className="bg-yellow"
														>
															{sendEmailMutation.isLoading ? <LoadingButton /> : 'Envoyer'}
														</Button>
													</Box>
												</Stack>
											</form>
										</>
									) : (
										<div className="flex flex-col gap-3 items-center">
											<Image src={"https://i.goopics.net/fuiyr8.png"} width={80} height={80} className="w-28" />
											<span className="text-xs text-gray-400 ">Un mail de réinitialisation a été envoyé à votre adresse email</span>
										</div>
									)
								}

							</div>
						</div>

					</div>
				</div>
			</div>
			{/* <BlinkSnackbar /> */}
		</>
	);
	return (
		<>
			{matches && !onboardingIsEnd ? (
				<OnboardingScreen
					onStartClick={() => {
						if (typeof window !== "undefined") {
							localStorage.setItem("onboardingIsEnd", "onboardingIsEnd");
						}
						setOnboardingIsEnd("onboardingIsEnd");
					}}
				/>
			) : (
				formHolder
			)}
		</>
	);
}

export default ForgotForm;
