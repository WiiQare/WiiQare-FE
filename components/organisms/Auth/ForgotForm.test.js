import React from "react";
import { render } from "@testing-library/react";
import ForgotForm from "./ForgotForm";
import { QueryClientProvider, QueryClient } from "react-query";

describe("ForgotForm", () => {
  it("renders", () => {
    const { container } = render(
      <QueryClientProvider client={new QueryClient()}>
        <ForgotForm />
      </QueryClientProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
