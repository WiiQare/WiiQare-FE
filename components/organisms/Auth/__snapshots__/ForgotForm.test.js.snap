// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForgotForm renders 1`] = `
<div>
  <div
    class="form-holder"
  >
    <div
      class="menu-holder"
    >
      <ul
        class="main-links"
      >
        <li>
          <div
            class="MuiBox-root css-1at62qq"
          >
            <a
              class="normal-link"
              href="/register"
            >
              Vous n'avez pas de compte ?
            </a>
          </div>
        </li>
        <li>
          <div
            class="MuiBox-root css-0"
          >
            <a
              href="/register"
            >
              <button
                class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium bg-yellow css-sghohy-MuiButtonBase-root-MuiButton-root"
                tabindex="0"
                type="button"
              >
                S'INSCRIRE
                <span
                  class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                />
              </button>
            </a>
          </div>
        </li>
      </ul>
    </div>
    <div
      class="signin-signup-form"
    >
      <div
        class="flex flex-col md:w-3/4 w-full border-opacity-50 gap-4"
      >
        <div
          class="form-items !w-full !max-w-full flex flex-col items-center mx-auto"
        >
          <div
            class="md:w-3/4 mx-auto"
          >
            <div
              class="form-title"
            >
              Mot de passe oublié
            </div>
            <div
              class="MuiBox-root css-19bt0nb"
            />
            <form
              class=""
              id="signinform"
            >
              <div
                class="css-1p5q5e5-MuiStack-root"
              >
                <div
                  class="space-y-1"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="outlined-basic"
                      id="outlined-basic-label"
                    >
                      Adresse email
                    </label>
                    <div
                      class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl css-md26zr-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-invalid="false"
                        class="MuiInputBase-input MuiOutlinedInput-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                        id="outlined-basic"
                        name="email"
                        type="text"
                        value=""
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-yjsfm1"
                        >
                          <span>
                            Adresse email
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  class="MuiBox-root css-0"
                >
                  <button
                    class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeLarge MuiButton-containedSizeLarge bg-yellow css-jh47zj-MuiButtonBase-root-MuiButton-root"
                    tabindex="0"
                    type="submit"
                  >
                    Envoyer
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
